### ISSUE TITLE
Compiler API should detect unparsed config files

### ISSUE THREAD

**Issue by RyanCavanaugh** [microsoft/TypeScript#27315]
Every week someone logs a bug that the compiler API "doesn't work" and posts something similar to this:
```ts
const config = JSON.parse(readFile("tsconfig.json"));
// config options not respected, why?
const program = createProgram(etc, config.compilerOptions);
```

In `createProgram` we should do a few `typeof` checks to ensure that e.g. `target`, `module`, `lib`, etc are not `string`, and throw a specific error:
```
 > createProgram was passed a raw config file - call ts.zzzzzzz to parse it first
```

e.g. #26622, #27268

**Comment by RyanCavanaugh** [comment]
OK it's not *that* common. We just need better docs.
